src/styles/getTooltipStyles.js

```diff
@@ -109,49 +108,54 @@ export default function getTooltipStyles(
     tooltipTargetWidth,
     tooltipTargetHeight,
     maxWidth,
-    tooltipWidth,
-    tooltipHeight,
     tooltipContentWidth,
+    tooltipContentHeight,
     manualShiftHorizontal = 0,
     manualShiftVertical = 0,
     tooltip,
 ) {
-    // Determine if the tooltip should display below the wrapped component.
-    // If either a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
-    // Or the wrapped component is overlapping at top-left with another element
-    // we'll display it beneath its wrapped component rather than above it as usual.
-    const shouldShowBelow = yOffset - tooltipHeight < GUTTER_WIDTH || isOverlappingAtTop(xOffset, yOffset, tooltip);
-
-    // Determine if we need to shift the tooltip horizontally to prevent it
-    // from displaying too near to the edge of the screen.
-    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, tooltipTargetWidth, tooltipWidth, manualShiftHorizontal);
-
-    // Determine if we need to shift the pointer horizontally to prevent it from being too near to the edge of the tooltip
-    // We shift it to the right a bit if the tooltip is positioned on the extreme left
-    // and shift it to left a bit if the tooltip is positioned on the extreme right.
-    const horizontalShiftPointer =
-        horizontalShift > 0
-            ? Math.max(-horizontalShift, -(tooltipWidth / 2) + POINTER_WIDTH / 2 + variables.componentBorderRadiusSmall)
-            : Math.min(-horizontalShift, tooltipWidth / 2 - POINTER_WIDTH / 2 - variables.componentBorderRadiusSmall);
-
     const tooltipVerticalPadding = spacing.pv1;
-    const tooltipFontSize = variables.fontSizeSmall;
-
-    // We calculate wrapper width based on the tooltip's inner text width so the wrapper is just big enough to fit text and prevent white space.
-    // If the text width is less than the maximum available width, add horizontal padding.
-    // Note: tooltipContentWidth ignores the fractions (OffsetWidth) so add 1px to fit the text properly.
-    const rootWrapperWidth = tooltipContentWidth && tooltipContentWidth + spacing.ph2.paddingHorizontal * 2 + 1;
 
-    const isTooltipSizeReady = tooltipWidth !== 0 && tooltipHeight !== 0;
+    // We calculate tooltip width and height based on the tooltip's content width and height
+    // so the tooltip wrapper is just big enough to fit content and prevent white space.
+    const tooltipWidth = tooltipContentWidth && tooltipContentWidth + spacing.ph2.paddingHorizontal * 2;
+    const tooltipHeight = tooltipContentHeight && tooltipContentHeight + tooltipVerticalPadding.paddingVertical * 2;
 
-    // Hide the tooltip entirely if it's size hasn't finished measuring yet. This prevents UI jank where the tooltip flashes close to its expected position.
-    const opacity = isTooltipSizeReady ? 1 : 0;
```

Comment:
Because now we depends on `isTooltipsizeReady` to set the opacity (previously xOffset and yOffset), we can safely remove it because the tooltip won't be shown if `useLayoutEffect` hasn't done yet..